core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckUpdateHandler.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import java.util.concurrent.CompletionStage;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.server.HttpResponseException;
+import com.linecorp.armeria.server.HttpStatusException;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Handles {@code PUT}, {@code POST} or {@code PATCH} requests sent to {@link HealthCheckService}.
+ */
+@FunctionalInterface
+public interface HealthCheckUpdateHandler {
+    /**
+     * Determines if the healthiness of the {@link Server} needs to be changed or not from the given
+     * {@link HttpRequest}.
+     *
+     * @return A {@link CompletionStage} which is completed with one of the following values:
+     *         <ul>
+     *           <li>{@code true} - the {@link Server} has to be marked as 'healthy'.</li>
+     *           <li>{@code false} - the {@link Server} has to be marked as 'unhealthy'.</li>
+     *           <li>{@code null} - the healthiness of the {@link Server} has to remain as-is.</li>
```

Comment:
How about using an enum instead of boolean? It's generally good if `null` an `false` have similar semantics for a `Boolean`

Reply:
Good idea. Replaced with `HealthCheckUpdateResult`.